packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -586,49 +569,42 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 			return;
 		}
 		assert(
-			entry.status === PendingBlobStatus.OnlinePendingUpload ||
-				entry.status === PendingBlobStatus.OfflinePendingUpload,
+			entry.status === PendingBlobStatus.Uploading,
 			0x386 /* Must have pending blob entry for uploaded blob */,
 		);
 		entry.storageId = response.id;
 		entry.uploadTime = Date.now();
 		entry.minTTLInSeconds = response.minTTLInSeconds;
+		// Send a blob attach op. This serves two purposes:
+		// 1. If its a new blob, i.e., it isn't de-duped, the server will keep the blob alive if it sees this op
+		//    until its storage ID is added to the next summary.
+		// 2. It will create a local ID to storage ID mapping in all clients which is needed to retrieve the
+		//    blob from the server via the storage ID.
+		if (!entry.opsent) {
+			this.sendBlobAttachOp(localId, response.id);
+		}
 		if (this.runtime.connected) {
```

Comment:
i don't think we should ever care about connected state in blob manager, i'm not sure why we check it here? if the blob upload resolved shouldn't we always do this?

Reply:
I think you're right. We are just avoiding going through the optimization path. After removing this if statement all tests work